static/app/components/globalSelectionLink.tsx

```diff
@@ -41,41 +42,41 @@ class GlobalSelectionLink extends React.Component<Props> {
       typeof to === 'object' && to.query ? {...globalQuery, ...to.query} : globalQuery;
 
     if (location) {
-      let toWithGlobalQuery;
+      let toWithGlobalQuery: LocationDescriptor;
+
       if (hasGlobalQuery) {
         if (typeof to === 'string') {
           toWithGlobalQuery = {pathname: to, query};
         } else {
           toWithGlobalQuery = {...to, query};
         }
       }
+
       const routerProps = hasGlobalQuery
-        ? {...this.props, to: toWithGlobalQuery}
+        ? {...this.props, to: toWithGlobalQuery!}
```

Comment:
with the above suggestion we could remove ` !`
```suggestion
        ? {...this.props, to: toWithGlobalQuery!}
```
```suggestion
        ? {...this.props, to: toWithGlobalQuery}
```